text/0000-project-ffi-unwind.md

```diff
@@ -0,0 +1,211 @@
+- Feature Name: `project-unwind-FFI`
+- Start Date: 2019-10-27
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+* To create a "project group" with the purpose of designing subsequent RFCs to
+  flesh out the details of the "C unwind" ABI
+  * The "project group" term is newly introduced: it is a specific type of
+    working group whose goal is to flesh out a particular proposal or complete
+    a project.
+  * This project group plans to specify how "C unwind" works on major
+    platforms.
+  * The primary goal is to enable Rust panics to propagate safely across
+    foreign frames.
+    * A future goal may be to enable foreign exceptions to propagate across Rust
+      frames.
+    * We do not plan to allow catching or throwing foreign exceptions from Rust
+      code
+
+# Motivation
+[motivation]: #motivation
+
+Unwinding through Rust's `extern "C"` ABI is [Undefined Behavior]. There is an
+[existing plan][abort-unwind] to make the behavior of Rust's `panic`
+well-defined by causing Rust functions defined with `extern "C"` to abort the
+application whenever an uncaught `panic` would otherwise escape into the
+caller. Unfortunately, previous attempts to stabilize this behavior have caused
+existing, working projects to break.
+
+The problem here is not that the existing projects break *per se*: they are
+relying on [Undefined Behavior], so breakage is to be expected as a
+possibility. The problem is that there is no alternative available to them that
+would allow them to keep working (even if they are continuing to rely on
+behavior that is not yet fully specified).
+
+Previous attempts to provide a well-defined mechanism for unwinding across FFI
+boundaries have failed to reach consensus. Notably, two proposed RFCs generated
+over 400 comments between them before ultimately being closed:
+
+* [rust-lang/rfcs#2699](https://github.com/rust-lang/rfcs/pull/2699)
+* [rust-lang/rfcs#2753](https://github.com/rust-lang/rfcs/pull/2753)
+
+GitHub comment threads become difficult to follow for discussions this lengthy,
+and the disagreements in these threads have felt less productive than we
+believe they could be if more structure were provided.
+
+We would also like to demonstrate the Rust lang team's commitment to providing
+such a mechanism without needing to agree in advance on what language changes
+will be stabilized in order to do so.
```

Comment:
```suggestion
will be stabilized in order to do so. Note however that the lang team's
commitment to some solution is not a commitment to *any* solution.
```
This should be obvious but I'd like to clarify this in the text nonetheless.

Reply:
I'm actually confused by this sentence; what do you mean by "any solution"? Are you just saying "we won't rush to stabilize a solution"?

Reply:
It just means that while we'll strive to find some acceptable solution, there may be solutions that are not acceptable.

Reply:
But I guess this can be left implicitly understood. :)